Content processing CI/CD is flaky

### This relates to ...

- [ ] the **FedRAMP OSCAL Registry**
- [ ] the **FedRAMP OSCAL baselines**
- [X] the **Guide to OSCAL-based FedRAMP Content**
- [ ] the **Guide to OSCAL-based FedRAMP System Security Plans (SSP)**
- [ ] the **Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP)**
- [ ] the **Guide to OSCAL-based FedRAMP Security Assessment Results (SAR)**
- [ ] the **Guide to OSCAL-based FedRAMP Plan of Action and Milestones (POA&M)**
- [ ] the **FedRAMP SSP OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP SAP OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP SAR OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP POA&M OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP OSCAL Validations**

### What happened?

It seems the content generation and testing CI in GHA is flaky looking at #641 failing on an unrelated change and clicking back through the `master` branch from time to time.

[Recent GHA job output](https://github.com/GSA/fedramp-automation/actions/runs/10615181568/job/29444121040?pr=641) leads me to one hypothesis, that is some parts of how to run Docker Compose has changed and the old way is no longer supported.

In the last few years, Docker Compose had a single wrapper command executable, `docker-compose` and then moved to `docker compose` (observe the space). It would seem that [GitHub discussion chatter confirms](https://github.com/orgs/community/discussions/27185) that was obsoleted.

This would mean local tests work, but not GHA. We need to investigate 1) if this is the only root case; 2) confirm/deny other root causes; 3) fix some/none/all and improve stability of all tests on PR and post-merge.

### Relevant log output

```Shell
##[debug]Evaluating condition for step: 'Build content generator'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Build content generator
##[debug]Loading inputs
##[debug]Loading env
Run make init-content
##[debug]/usr/bin/bash -e /home/runner/work/_temp/4b01530c-f811-45dd-82e9-e2bd38a89b21.sh
Building Docker image for OSCAL content generation...
docker-compose -f vendor/oscal/build/docker-compose.yml -f src/content/docker-compose.yml build
make: docker-compose: Command not found
make: *** [src/content/module.mk:16: init-content] Error 127
Error: Process completed with exit code 2.
##[debug]Finishing: Build content generator
```


### How do we replicate this issue?

1. Create a PR with any change.
2. Observe somewhat consistent failure of the `content-artifacts.yml` workflow.

### Where, exactly?

GitHub Actions, so `.github/workflows`.

### Other relevant details

_No response_

--- a/.github/workflows/content-artifacts.yml
+++ b/.github/workflows/content-artifacts.yml
@@ -20,7 +20,7 @@ jobs:
     name: Content Validation Checking, Conversion and Validation
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
         with:
           path: git-content
           submodules: recursive

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -9,7 +9,7 @@ jobs:
 
     steps:
       # Check-out the repository under $GITHUB_WORKSPACE
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
 
       - name: Read Node version from .nvmrc
         run: echo "##[set-output name=NVMRC;]$(cat .nvmrc)"

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -24,7 +24,7 @@ jobs:
     steps:
       # Check-out the repository under $GITHUB_WORKSPACE
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
 
       - name: Read node version from `.nvmrc` file
         id: nvmrc

--- a/.github/workflows/runSchematronValidation.yml
+++ b/.github/workflows/runSchematronValidation.yml
@@ -24,7 +24,7 @@ jobs:
     # Checkout repository and its submodules
     steps:
       # Check-out the repository under $GITHUB_WORKSPACE
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
 
       # Check-out submodules
       - name: Checkout submodules

--- a/src/content/docker-compose.yml
+++ b/src/content/docker-compose.yml
@@ -1,6 +1,5 @@
 # This extends the OSCAL build container which should be located at:
 # ../../vendor/oscal/build/docker-compose.yml
-version: '3'
 services:
   cli:
     volumes:

--- a/src/content/module.mk
+++ b/src/content/module.mk
@@ -9,7 +9,7 @@ OSCAL_DIR := vendor/oscal
 CICD_DIR_PATH := $(OSCAL_DIR)/build/ci-cd
 CONTENT_CONFIG_PATH := src/config
 
-OSCAL_DOCKER := docker-compose -f $(OSCAL_DIR)/build/docker-compose.yml -f $(CONTENT_DIR)/docker-compose.yml
+OSCAL_DOCKER := docker compose -f $(OSCAL_DIR)/build/docker-compose.yml -f $(CONTENT_DIR)/docker-compose.yml
 
 init-content:  ## Initialize content build environment (build Docker image)
 	@echo "Building Docker image for OSCAL content generation..."

--- a/src/examples/java/README.md
+++ b/src/examples/java/README.md
@@ -15,14 +15,14 @@ make build-validations
 
 ## Usage
 
-This example uses Maven. You may run directly, or via the provided `docker-compose` configuration.
+This example uses Maven. You may run directly, or via the provided `docker compose` configuration.
 
 ```bash
 mvn --help
 ```
 
 ```bash
-docker-compose run example mvn --help
+docker compose run example mvn --help
 ```
 
 ## Developer notes
@@ -30,11 +30,11 @@ docker-compose run example mvn --help
 To run tests:
 
 ```bash
-docker-compose run example mvn test
+docker compose run example mvn test
 ```
 
 To auto-format code:
 
 ```bash
-docker-compose run example mvn com.coveo:fmt-maven-plugin:format
+docker compose run example mvn com.coveo:fmt-maven-plugin:format
 ```

--- a/src/examples/javascript/README.md
+++ b/src/examples/javascript/README.md
@@ -23,7 +23,7 @@ make build-validations
 To get a bash prompt within the container, run:
 
 ```bash
-docker-compose run example bash -l
+docker compose run example bash -l
 ```
 
 You may now run and interact with the example code.
@@ -33,11 +33,11 @@ You may now run and interact with the example code.
 To run tests, build the required Schematron XSLT to sef.json files and run the test script:
 
 ```bash
-docker-compose run example npm run ./test.sh
+docker compose run example npm run ./test.sh
 ```
 
 To auto-format code:
 
 ```bash
-docker-compose run example npm format
+docker compose run example npm format
 ```

--- a/src/examples/module.mk
+++ b/src/examples/module.mk
@@ -3,14 +3,14 @@ test-examples: test-example-java test-example-python test-example-javascript  ##
 test-example-java:  ## Test example Java project
 	@echo "Verifying Java example..."
 	cd src/examples/java && \
-		docker-compose run example mvn test
+		docker compose run example mvn test
 
 test-example-python:  ## Test example Python project
 	@echo "Verifying Python example..."
 	cd src/examples/python && \
-		docker-compose run example pytest
+		docker compose run example pytest
 
 test-example-javascript:  ## Test example Javascript project
 	@echo "Verifying Javascript example..."
 	cd src/examples/javascript && \
-		docker-compose run example ./test.sh
+		docker compose run example ./test.sh

--- a/src/examples/python/README.md
+++ b/src/examples/python/README.md
@@ -19,7 +19,7 @@ make build-validations
 To get a bash prompt within the container, run:
 
 ```bash
-docker-compose run example bash -l
+docker compose run example bash -l
 ```
 
 You may now run and interact with the example code.
@@ -29,17 +29,17 @@ You may now run and interact with the example code.
 To run tests:
 
 ```bash
-docker-compose run example pytest
+docker compose run example pytest
 ```
 
 To auto-format code:
 
 ```bash
-docker-compose run example black
+docker compose run example black
 ```
 
 To type-check code:
 
 ```bash
-docker-compose run example mypy .
+docker compose run example mypy .
 ```

--- a/src/validations/CONTRIBUTING.md
+++ b/src/validations/CONTRIBUTING.md
@@ -59,10 +59,10 @@ Using the command with a different version of Saxon HE. *Note that you must down
 ./bin/validate_with_schematron.sh -f test/demo/FedRAMP-SSP-OSCAL-Template.xml -o ~/dev/report -v 10.8
 ```
 
-Alternatively, you can also use `docker-compose` to execute the validation script, like so:
+Alternatively, you can also use `docker compose` to execute the validation script, like so:
 
 ```sh
-docker-compose run \
+docker compose run \
   -w /root/src/validations \
   validator \
   ./bin/validate_with_schematron.sh -f ./test/demo/FedRAMP-SSP-OSCAL-Template.xml
@@ -100,10 +100,10 @@ export TEST_DIR=$(pwd)/report/test
 
 The JUnit XML report will be stored in the path set in the `TEST_DIR` environment variable.
 
-Alternatively, you can also use `docker-compose` to execute the test harness, like so:
+Alternatively, you can also use `docker compose` to execute the test harness, like so:
 
 ```sh
-docker-compose run \
+docker compose run \
   -w /root/src/validations \
   validator \
   /root/vendor/xspec/bin/xspec.sh -s -j ./test/test_all.xspec

